dts/arm/nxp/nxp_rt6xx_common.dtsi

```diff
@@ -48,6 +49,20 @@
 	#address-cells = <1>;
 	#size-cells = <1>;
 
+	clkctl0: clkctl@1000 {
+		compatible = "nxp,lpc-syscon";
+		reg = <0x1000 0x4000>;
+		label = "CLKCTL_0";
+		#clock-cells = <1>;
+	};
+
+	clkctl1: clkctl@21000 {
+		compatible = "nxp,lpc-syscon";
```

Comment:
This should be something like `compatible = "nxp,rt6xx-syscon";`

Reply:
RT600 uses a lot of LPC IP blocks such as SYSCON, FLEXCOMM, GPIO etc. This is meant to be available for all SoC's such as RT600, LPC55S69, LPC54114 etc.

Reply:
Sure, the idea with the `compatible` is it describes the programming model of the hardware.  Since the register layouts of SYSCON are vastly different between RT6xx and LPC5xxx I made my comment about the `compatible` needing to be different.

If I look at the UM for LPC5411x and LPC55S69 there are numerous registers at the some offsets with the same purpose.  For example FCCLKSEL0 is at offset 0x2B0.  On RT6xx the register is called FC0FCLKSEL and at offset 0x508.

Reply:
All hardware variations are handled by the SDK clock driver. The SDK clock driver API's are what is used by the zephyr clock-control driver.